### DISCUSSION TITLE
Soketi Integration with Pusher in Node Js

### DISCUSSION THREAD

**Discussion by vivekkotadiya** [soketi/pws#1007] [Q&A]
I am new to Soketi and I want to integrate it into one site where we are using Pusher. But due to the limit in concurrent users and messages we are planning to migrate to Soketi. Currently, we are building an app in Node Js and Next Js (Frotn-end) so need an integration guide here. If we can use Soketi with Pusher how can we manage unlimited concurrent connections and messages? I tried to Pusher package for Node Js ingration and when I run it's triggers even if it counts messages and connections in the Pusher dashboard. So I think it will restrict messages and connections based on activity. Also, in the Soketi app config which app ID, secret key etc should be used?

Can you guide me for that how I can integrate it with node js for triggering events from Soketi using Pusher? Also, how I can handle concurrent connection with horizontal scaling? I am using Cloudways hosting as of now which is using Digital Ocean droplet.

Here is my demo code sample.
https://prnt.sc/r1ODgqFw85Ev
https://prnt.sc/ScVt9zXh44IS

I tested my connection with Soketi from Postman and it is showing connected.
https://prnt.sc/psF-_EP87tfG

**Comment by aminegazzeh** [comment]
@vivekkotadiya did you find a solution for your problem ?

**Comment by vivekkotadiya** [comment]
@aminegazzeh Yes, There is simple solutions for use Pusher with Soketi with any credentials like below:

```
let pusher = new Pusher({
    appId: 'my app id', // Any random key you can use
    key: 'My app key', // Any random key
    secret: 'My secret', // Any random key
    useTLS: true, // If used SSL
    scheme: 'https', // If SSL used
    host: 'YOUR_HOST', // ex.: https://mydomain.com
    port: 'PORT' // Port number on which you node app is running
});
```

For event trigger: ```pusher.trigger();```

On the frontend HTML bind pusher js file like: `<script src="https://js.pusher.com/7.0/pusher.min.js"></script>`

```
const pusher = new Pusher('my app key' // Any random key but should be same as app key used in node js,
{
	cluster: 'eu-west-1',
	wsHost: 'YOUR_NODE_APP_HOST', // Used non SSL site
	wsPort: 6002,  // Used non SSL site
	wssHost: 'YOUR_NODE_APP_HOST',  // Used SSL site
	wssPort: 6002,
	forceTLS: true,  // Used SSL site
	enabledTransports: ['ws', 'wss'],
});
```
